### ISSUE TITLE
Update editor to allow display and edit licenses for figures

### ISSUE THREAD

**Issue by NuclearRedeye** [libero/editor#577]
[Design](https://projects.invisionapp.com/d/main#/console/19776007/433142658/preview)

# DoD
- [ ] User can view the licenses for a specific figure as per the design.
- [ ] User can using the edit affordance open the edit modal to make a change to a license.
- [ ] User can delete a license.
- [ ] User can add a new license using the 'add license' affordance.
- [ ] User can change the type of license, and the available options in the modal dialog should change (similar to references) as per the design.

# Design proposal
- [Adding licenses design spec](https://invis.io/A8WXRN9PQBF#/433142658_figure-2). Click add license link
- [License types display](https://invis.io/A8WXRN9PQBF#/430713004_figure-3)  

Note data source is shown in design page 4 but not required for mvp

**Comment by fred-atherden** [comment]
1. The first item in the license type dropdown says `CCO`, when it should say `CC0` (it's a zero not an o).

![Screenshot 2020-11-09 at 15.21.56.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/ca26ec79-c6fc-49c9-b0fc-c25b72bb5fab)

2. Please can we change the second item in the license type dropdown to say `CC by 4.0`, as per [the design](https://projects.invisionapp.com/share/A8WXRN9PQBF#/screens/430713004).

![Screenshot 2020-11-09 at 15.21.56.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/d2d32216-46b0-459c-b1c7-6fc4020de1b6)

3. The attribution is currently displaying in the License field. Unsure whether that's just because attribution hasn't been implemented yet, but it's incorrect.
![Screenshot 2020-11-09 at 15.25.55.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/538d033a-07c7-4b53-827d-b52123af9907)

4. ~~There's an extra License statement field. I don't think this is needed (at least it's not in the design) but @NickDuf can confirm.~~
[Edit: above text is wrong. There is a license statement field. Looks like it needs to appear above the License field instead though.]
![Screenshot 2020-11-09 at 15.25.55.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/3530daea-1956-441d-8d10-1dd5b6187096)

5. Adding any formatting (such as bold, italic, hyperlinks) to text in the License field takes me to the top of the Figure (I lose my position on the page).
**Steps to reproduce**
a. Visit here -https://libero-editor--staging--test-cluster.elifesciences.org/?articleId=54296
b. Navigate to the license for Figure 1.
c. Add text to License field and add bold formatting (or if text is already present, then simply format some of it bold).
d. Note that on click, you are taken to the top of the Figure.
Happens with toolbar and hotkeys.
 
6. When inserting a reference citation in the labels for the other license fields appear in the pop-up. Here's an example of the year label appearing:
![Screenshot 2020-11-09 at 15.34.33.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/8558b41a-1b42-4938-afca-cce00e4f714e)

(Note you have to be sufficiently scrolled so that the reference pop-up appears on top of the field(s).)

7. It takes three attempts to add two reference citations in the license field. Adding one reference citation works fine. Adding the second means that you have to click the affordance twice.
**Steps to reproduce**
a. Visit here -https://libero-editor--staging--test-cluster.elifesciences.org/?articleId=54296
b. Navigate to the license for Figure 1.
c. Insert one reference citation using the toolbar. Note that it works as expected.
d. Insert a second reference using the toolbar. Note that on click the `???` does not appear. 
e. Attempting this again works.

**Comment by fred-atherden** [comment]
Inserting three reference citations means that the second is changed back to `???`. 

**Steps to reproduce**
a. Visit here -https://libero-editor--staging--test-cluster.elifesciences.org/?articleId=54296
b. Navigate to the license for Figure 1.
c. Insert one reference citation using the toolbar. Note that it works as expected.
d. Insert a second reference using the toolbar. Note that on click the ??? does not appear.
e. Attempting this again works.
f. Insert a third reference citation using the toolbar. Note that the second citation is now changed back to `???`:
![Screenshot 2020-11-09 at 15.45.55.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/7c8dda92-fa97-4d3c-ba82-15f34747cfc5)

**Comment by fred-atherden** [comment]
If there are 4+ licenses, then if you delete a license before the 4th, the text of the 4th license is changed.

Steps to reproduce
a. Visit here -https://libero-editor--staging--test-cluster.elifesciences.org/?articleId=54296
b. Navigate to the license for Figure 1.
c. Add three licenses so that there are 4 total.
d. Add the text 'Test {license number}' in the license field for Licenses 2-4.
![Screenshot 2020-11-09 at 15.50.12.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/41ec2bd4-a5c0-44e5-bd2f-dbddf1b38100)
![Screenshot 2020-11-09 at 15.50.16.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/777cd46a-41a9-4f9d-91fe-c47b7ead17ba)
![Screenshot 2020-11-09 at 15.50.30.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/9ba565c2-e539-4e1f-b075-47d10e73ba2b)
e. Delete the second license.
f. Note that both remaining licenses (what were 3 and 4, but are now 2 and 3) display 'Test 3' in the License field:
![Screenshot 2020-11-09 at 15.51.38.png](https://images.zenhubusercontent.com/5d5a77bfe56ed603cbb191f6/182ba9e7-a71b-427c-bba3-76d54cc7fbdb)

**Comment by NickDuf** [comment]
@FAtherden-eLife @ilya-zhuravel has matched the design for your point 4. I am happy for the change to be made that you suggest though.

Point 5: I have the same issues.

My minor feedback
- License 1 input in it's default needs to match min-height of other text area inputs found on things like the bio in the author card
- Padding top within the license card to be 20px to visually match the designs

**Comment by fred-atherden** [comment]
> @FAtherden-eLife @ilya-zhuravel has matched the design for your point 4. I am happy for the change to be made that you suggest though.

Thanks @NickDuf, looks like it's already been changed if I compare my screenshot above with Editor now. Quick work @ilya-zhuravel!

**Comment by fred-atherden** [comment]
@ilya-zhuravel, thanks it looks like all of the problems are now resolved with the exception of this one - https://github.com/libero/editor/issues/577#issuecomment-724099500. I'm still seeing this behaviour.

One other problem I've just noticed is that in the license field, the hotkeys are now acting a little strangely. It takes two `cmd + b`s for me to bold and 5 to unbold. Same with the hotkeys for italic and underline.
